verifiable/onchain/parser.go

```diff
@@ -0,0 +1,473 @@
+package onchain
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	core "github.com/iden3/go-iden3-core/v2"
+	"github.com/iden3/go-iden3-core/v2/w3c"
+	"github.com/iden3/go-merkletree-sql/v2"
+	"github.com/iden3/go-schema-processor/v2/merklize"
+	"github.com/iden3/go-schema-processor/v2/verifiable"
+	"github.com/piprate/json-gold/ld"
+)
+
+type Parser struct {
+	marklizeInstance merklize.Options
+
+	regenerateID bool
+}
+
+type Option func(*Parser)
+
+// WithRegenerateID sets the regenerateID flag to the given value.
+func WithRegenerateID(regenerateID bool) Option {
```

Comment:
WithRegenerateID -> withIDGeneration / withID, so you can generate id how do you want

Reply:
@vmidyllic We have some rules for ID. It should be uri. If the user creates an id without the prefix `urn:uuid:` or something similar, proof generation will not work I sink.

Reply:
You can add validation for Uri. any url or any other uri format other then urn:uuid is valid